providers/provider_default.go

```diff
@@ -40,7 +43,7 @@ func (p *ProviderData) GetLoginURL(redirectURI, state, _ string) string {
 }
 
 // Redeem provides a default implementation of the OAuth2 token redemption process
-func (p *ProviderData) Redeem(ctx context.Context, redirectURL, code string) (*sessions.SessionState, error) {
+func (p *ProviderData) Redeem(ctx context.Context, redirectURL, code string, idString string) (*sessions.SessionState, error) {
```

Comment:
Should the ID of the provider not be inserted within the ProviderData already? The providerID won't change during execution so can be stored in the struct at the time the provider is initialised right?

Reply:
Yes I think this is correct - the idString is already avaliable at p.ProviderID so is not needed to be passed again here.